### PR TITLE
fix RESOLVE_URIS_COMPLETED, to have one large action instead of hundr…

### PR DESCRIPTION
**Pull request by AsadUmar**
## Fixes

Issue Number: https://github.com/OdyseeTeam/odysee-frontend/issues/727

## What is the current behavior?

On http://localhost:9090/$/list/d91815c1bc8c80a1f354284a8c8e92d84d5f07a6 5x claim_search and 193x resolve requests are made.

## What is the new behavior?

On http://localhost:9090/$/list/d91815c1bc8c80a1f354284a8c8e92d84d5f07a6 5x claim_search requests are made.

## Other information

Nothing crashes as far as I've tested but probably needs a review by someone who knows whats what. 

I didn't write much of this it was basically taken from `ui/redux/actions/claims.js` and I made sure that the data mapping made sense in redux.

## PR Checklist

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change (Could be wrong though)
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

### CODE REVIEW

**Review by infinite-persistence — CHANGES_REQUESTED** [review]
- If the mistake is in `formatForClaimActions`, it should be fixed there rather than duplicated below and end up double parsing.
    - I didn't look too deep, but doing `collectionItemsById[0].items` feels like we will just handle 1 collection.  When loading "my collections" page, the variable could hold N collections.
    - The `map` feels like an unnecessary extra loop?
    - Move `processResult` outside of the loop?
- But the issue still remains: the `ClaimPreview` will still end up calling a `resolve` because  it is mounted before this part of the code finishes.  
    - It seems to be already checking `resolvingUris` before resolving, so maybe that flag isn't updated correctly for the case of collections.

**Comment by AsadUmar** [comment]
>   * I didn't look too deep, but doing `collectionItemsById[0].items` feels like we will just handle 1 collection.  When loading "my collections" page, the variable could hold N collections.

Agreed (though in my experience collectionItemsById almost always has one element but better safe than sorry)

>   * The `map` feels like an unnecessary extra loop?

True, I can change that, but the reason behind doing it this way is so that if need be later on `processResult` could be shared between claims.js and collections.js.

>   * Move `processResult` outside of the loop?

Sure.

> * But the issue still remains: the `ClaimPreview` will still end up calling a `resolve` because  it is mounted before this part of the code finishes.

AFAIK, thats probably the right way to go as if a claim does not resolve in `doFetchItemsInCollections` it would get resolved  later on when the component is mounted.
   
>   * It seems to be already checking `resolvingUris` before resolving, so maybe that flag isn't updated correctly for the case of collections.

not sure what you mean there.

**Review by infinite-persistence — CHANGES_REQUESTED** [review]
Thanks for the update.

The resolve problem is still open.  Considering the individual tiles are mounted before the fetch (thus launching the fallback `resolve` calls)...

image

... the solution would probably to not mount `ClaimList` in `collection/view.jsx` until we are done with `fetchCollectionItems`.

File: ui/redux/actions/collections.js [hunk 1]
```diff
@@ -289,11 +289,36 @@ export const doFetchItemsInCollections = (
...
+
   dispatch({
     type: ACTIONS.RESOLVE_URIS_COMPLETED,
     data: { resolveInfo: formattedClaimsByUri },
   });
 
+  dispatch({
+    type: ACTIONS.RESOLVE_URIS_COMPLETED,
+    data: { resolveInfo },
+  });
```
[the remark this answers is not in the document]
Comment (AsadUmar) [reply ui/redux/actions/collections.js]: Fixed

File: ui/redux/actions/collections.js [hunk 2]
```diff
@@ -289,11 +289,36 @@ export const doFetchItemsInCollections = (
...
   const formattedClaimsByUri = formatForClaimActions(collectionItemsById);
 
+  const resolveInfo: {
+    [string]: {
+      stream: ?StreamClaim,
+      channel: ?ChannelClaim,
+      claimsInChannel: ?number,
+      collection: ?CollectionClaim,
+    },
+  } = {};
```
[the remark this answers is not in the document]
Comment (AsadUmar) [reply ui/redux/actions/collections.js]: Fixed

**Comment by tzarebczan** [comment]
> 

Was this resolved? Not sure. Take a peek tomorrow when you're up.

**Pull request merged**